### DISCUSSION TITLE
Why does displacy raise a warning when rendering Docs parsed using a Stanza language model via spacy-stanza?

### DISCUSSION THREAD

**Discussion by thiippal** [explosion/spaCy#7354] [Help: Other Questions] [answered]
Hey!

I used spacy-stanza to load a Stanza language model for Finnish into spaCy.

I then applied the model to some texts to get a Doc object. I then used the `sents` attribute to extract Span objects for sentences, which I stored into a list named `sents_fi`.

I then wanted to visualise the syntactic dependencies for the first Span in the list.

```python
# Import the displacy submodule
from spacy import displacy

# Use the render function to render the first item [0] in the list 'sents_fi'.
# Pass the argument 'style' with the value 'dep' to visualise syntactic dependencies.
displacy.render(sents_fi[0], style='dep')
``` 

This works just fine, but raises the following warning:

```
[W109] Unable to save user hooks while serializing the doc. Re-add any required user hooks to the doc after processing.
```

Any idea what causes this warning?

**Comment by adrianeboyd** [comment, marked as answer]
This is a relatively cryptic error that you can safely ignore or filter if you're just using displacy. I think we should consider filtering it out ourselves with displacy, since it's really confusing.

What's going on underneath in case you're interested: `spacy-stanza` add user hooks for token vectors if the stanza model includes pretrained word embeddings. As an internal processing step, displacy serializes the doc with `doc.to_bytes`, which is what leads to this warning, since it can't serialize the user hooks.

**Reply by adrianeboyd** [reply]
Should be fixed by #7419.
